openstack/orchestration/v1/stacks/template.go

```diff
@@ -76,15 +78,15 @@ func (t *Template) getFileContents(te interface{}, ignoreIf igFunc, recurse bool
 
 				// initialize child template
 
-				// get the base location of the child template
-				baseURL, err := gophercloud.NormalizePathURL(t.baseURL, value)
-				if err != nil {
-					return err
+				// get the base location of the child template. Child path is relative
+				// to its parent location so that templates can be composed
+				if t.URL != "" {
+					childTemplate.baseURL = filepath.Dir(t.URL)
```

Comment:
@dverbeir since we don't use `NormalizePathURL` anymore have you checked that we don't let URL escape here?

Reply:
In fact, the `childTemplate.Fetch()` a few lines below will invoke `NormalizePathURL()`. So I think it's safe.

Also, I tried using `NormalizePathURL()` here but when I do, and I see the correct resulting baseURL, the second invocation inside `Fetch()` messes it up, doubling parts of the path...

TBH It's totally unclear to me what the intention was with `baseURL` for templates. I see it's more used for the "templates" inside `Environment` but I'm not familiar with the workings of that part.